jenkins_jobs/modules/publishers.py

```diff
@@ -2456,37 +2456,119 @@ def base_email_ext(registry, xml_parent, data, ttype):
     trigger = XML.SubElement(
         xml_parent, "hudson.plugins.emailext.plugins.trigger." + ttype
     )
+
+    info = registry.get_plugin_info("email-ext")
+    plugin_version = pkg_resources.parse_version(info.get("version", str(sys.maxsize)))
+
     email = XML.SubElement(trigger, "email")
-    XML.SubElement(email, "recipientList").text = ""
+
+    if plugin_version < pkg_resources.parse_version("2.39"):
+        XML.SubElement(email, "recipientList").text = ""
     XML.SubElement(email, "subject").text = "$PROJECT_DEFAULT_SUBJECT"
     XML.SubElement(email, "body").text = "$PROJECT_DEFAULT_CONTENT"
+    if plugin_version >= pkg_resources.parse_version("2.39"):
+        XML.SubElement(email, "replyTo").text = "$PROJECT_DEFAULT_REPLYTO"
+        XML.SubElement(email, "contentType").text = "project"
     if "send-to" in data:
-        XML.SubElement(email, "sendToDevelopers").text = str(
-            "developers" in data["send-to"]
-        ).lower()
-        XML.SubElement(email, "sendToRequester").text = str(
-            "requester" in data["send-to"]
-        ).lower()
-        XML.SubElement(email, "includeCulprits").text = str(
-            "culprits" in data["send-to"]
-        ).lower()
-        XML.SubElement(email, "sendToRecipientList").text = str(
-            "recipients" in data["send-to"]
-        ).lower()
+        if plugin_version < pkg_resources.parse_version("2.39"):
+            XML.SubElement(email, "sendToDevelopers").text = str(
+                "developers" in data["send-to"]
+            ).lower()
+            XML.SubElement(email, "sendToRequester").text = str(
+                "requester" in data["send-to"]
+            ).lower()
+            XML.SubElement(email, "includeCulprits").text = str(
+                "culprits" in data["send-to"]
+            ).lower()
+            XML.SubElement(email, "sendToRecipientList").text = str(
+                "recipients" in data["send-to"]
+            ).lower()
         if "upstream-committers" in data["send-to"]:
             recipient_providers = XML.SubElement(email, "recipientProviders")
             XML.SubElement(
                 recipient_providers,
                 "hudson.plugins.emailext.plugins.recipients.UpstreamComitterRecipientProvider",
             ).text = ""
+        if plugin_version >= pkg_resources.parse_version("2.39"):
+            for recipient in data["send-to"]:
+                if "developers" == recipient:
+                    if "recipient_providers" not in locals():
```

Comment:
I'd rather set `recipient_providers` to `None` at the beginning of the whole code and check the variable for `None` later on. Using `locals()` is for such cases is rather a bad practice.